src/dialogflow/sessions_client.rs

```diff
@@ -54,8 +54,8 @@ impl SessionsClient {
 
     /// Returns sender than can be used to stream in audio bytes.
     pub fn get_audio_sink(&mut self) -> Option<mpsc::Sender<StreamingDetectIntentRequest>> {
-        if let Some(audio_sender) = &self.audio_sender {
-            Some(audio_sender.clone())
+        if let Some(audio_sender) = self.audio_sender.take() {
```

Comment:
when this is used error will be something like (**message: "status: Unknown, message: \"grpc-status header missing, mapped from HTTP status code 200\"**)

which is better but in ideal case this should end gracefully without any error. which probably will not be possible until we can send **half close** signal into stream, right? From what I remember once you get timeout error the stream is basically broken and you cannot use it any more to send additional data. in that case whole stream must be recreated. 

I am still thinking if it can be useful in some cases just clone the sender instead of yanking it from the `Option `with `take `as you do. Nothing comes to my mind. Any ideas whether it would make sense to have for example to `get_audio_sink` versions (one with `clone `the other one with `take`)?

Reply:
I wouldn't mind the instance where you can have the cloned and the dropped version. Or maybe also a function which just takes and drops the one in the option so once you've dropped all the cloned ones you can clear it.

I looked into the stream half close with the 'end stream' flag with gRPC and tonic. I think you should see if these errors persist with the version bump to tonic. 

We're using this project but we have a fork, so no rush or pressure with any of these changes - it was just what was required to get it working for us.

Reply:
ok, let's do it like this:

1. let's keep the original **get_audio_sink** method that is simply cloning the sender. The user has then possibility to receive even multiple copies should it make sense in some very specific scenario.
2. let's add the method **drop_audio_sink** that will simply take the sender from the Option and drop it
3. maybe we could also add convenience method **take_audio_sink** which would call take instead of clone (or simply call internally get_audio_sink followed by take_audio_sink).

this  way it will be flexible for different use cases. Does this make sense?